scripts/execute_ebpf_cicd_tests.ps1

```diff
@@ -1,86 +1,114 @@
-    Invoke-XDPTestsOnVM `
-        -Interfaces $Config.Interfaces `
-        -VMName $VMList[0].Name `
-        -TestHangTimeout $TestHangTimeout `
-        -UserModeDumpFolder $UserModeDumpFolder
-
-    # Run Connect Redirect Tests.
-    Invoke-ConnectRedirectTestsOnVM `
-        -Interfaces $Config.Interfaces `
-        -ConnectRedirectTestConfig $Config.ConnectRedirectTest `
-        -UserType "Administrator" `
-        -VMName $VMList[0].Name `
-        -TestHangTimeout $TestHangTimeout `
-        -UserModeDumpFolder $UserModeDumpFolder
-
-    Invoke-ConnectRedirectTestsOnVM `
-        -Interfaces $Config.Interfaces `
-        -ConnectRedirectTestConfig $Config.ConnectRedirectTest `
-        -UserType "StandardUser" `
-        -VMName $VMList[0].Name `
-        -TestHangTimeout $TestHangTimeout `
-        -UserModeDumpFolder $UserModeDumpFolder
-}
+    Push-Location $WorkingDirectory
+
+    # Load other utility modules.
+    Import-Module $WorkingDirectory\common.psm1 -Force -ArgumentList ($LogFileName) -WarningAction SilentlyContinue
+    Import-Module $WorkingDirectory\vm_run_tests.psm1 `
+        -Force `
+        -ArgumentList (
+            $Admin,
+            $AdminPassword,
+            $StandardUser,
+            $StandardUserPassword,
+            $WorkingDirectory,
+            $LogFileName,
+            $TestMode,
+            $Options,
+            $TestHangTimeout,
+            $UserModeDumpFolder) `
+        -WarningAction SilentlyContinue
+
+    # Run Kernel tests on test VM.
+    Write-Log "Running kernel tests on $TestVMName"
+    Run-KernelTestsOnVM -VMName $TestVMName -Config $Config
 
-# Stop eBPF components on test VMs.
-foreach ($VM in $VMList) {
-    Stop-eBPFComponentsOnVM -VMName $VM.Name
+    # Stop eBPF components on test VMs.
+    Stop-eBPFComponentsOnVM -VMName $TestVMName
+} -ArgumentList (
+    $TestVMName,
+    $Config,
+    $AdminTestVMCredential.UserName,
+    $AdminTestVMCredential.Password,
+    $StandardUserTestVMCredential.UserName,
+    $StandardUserTestVMCredential.Password,
+    $WorkingDirectory,
+    $LogFileName,
+    $TestMode,
+    $Options,
+    $TestHangTimeout,
+    $UserModeDumpFolder)
+
+# Keep track of the last received output count
+$LastCount = 0
+$TimeElapsed = 0
+
+# Loop to fetch and print job output in near real-time
+while ($Job.State -eq 'Running') {
+    $JobOutput = Receive-Job -Job $job
+	$JobOutput | ForEach-Object { Write-Host $_ }
+
+    Start-Sleep -Seconds 2
+    $TimeElapsed += 2
+
+    if ($TimeElapsed -gt $TestJobTimeout) {
+        if ($Job.State -eq "Running") {
+            Write-Host "Running kernel tests on $TestVMName has timed out after one hour" -ForegroundColor Yellow
+            $Timestamp = Get-Date -Format 'yyyy-MM-dd_HH-mm-ss'
+            $CheckpointName = "$TestVMName-Checkpoint-$Timestamp"
+            Write-Log "Taking snapshot $CheckpointName of $TestVMName"
+            Checkpoint-VM -Name $TestVMName -SnapshotName $CheckpointName
```

Comment:
When does the while loop terminate? Is it possible that this starts to create many checkpoints?